src/Nfts/ImageGallery/ImageGallery.tsx

```diff
@@ -58,81 +88,159 @@ interface ModeratedImageProps {
 
 const ModeratedImage = ({fingerprint, image, text, onPress}: ModeratedImageProps) => {
   const wallet = useSelectedWallet()
-  const moderationStatusQuery = useQuery({
-    queryKey: [wallet.id, 'nft', fingerprint],
-    queryFn: () => wallet.fetchNftModerationStatus(fingerprint),
-  })
+  const moderationStatusQuery = useNftModerationStatus({wallet, fingerprint})
 
   useEffect(() => {
```

Comment:
this looks suspect, what is this doing?

Reply:
this should part of NFTImage

Reply:
@wolverineks Do you mean fetching the moderation status? Based on the status (pending, approved, consent, blocked) we display the image/ blurred image/ placeholder.

@stackchain   at the moment it's in `ModeratedImage`. Where would you expect this to be?

Reply:
`refetchInterval(({ moderationStatus }) => moderationStatus === "pending" ? REFETCH_TIME_IN_MS : false)`